### ISSUE TITLE
About DrawFilter.py

### ISSUE THREAD

**Issue by BassyKuo** [ShihHsuanChen/mnist_analysis#1]
In line 60: https://github.com/ShihHsuanChen/mnist_analysis/blob/master/DrawFilter.py#L60
Here is my understanding of your code:
![fullsizerender 13](https://user-images.githubusercontent.com/6426781/30784414-3b4c6128-a187-11e7-8d3f-59d5751e8c86.jpg)
It seems that you want to show the filter pattern, but you scale up the filter value by multiplying the input pixels even unmatching the filter index? (e.g. `W_30 * input_02`) That is a bit strange.

Furthermore, because your original input image is between [-1,1]-pixel-value (do you?), the filter would be sensitive to negative pixels masked the negative filter value (i.e. assumed `tmp_weight = W_00*input_00`, if `W_00 < 0 and input_00 < 0` then `tmp_weight > 0`). If you take `tmp_pict` to be the weighting, the `tmp_pict[i,j] * tmp_weight` would be the negative value if `tmp_pict[i,j]` is negative, it would cause `tmp_pict[i,j] * tmp_weight` to be the minimum in your FilterPolts. So the FilterPolt you see may not be the truth.

Maybe you can try to shift the `W_conv1` to make the minimum to be zero, and then scale the value to be in the range [0,255]. It might show the ugly, ununderstanding but true filter pattern.
Or there are other ways to visualize filters by gradient computing: [[ref1]](https://jacobgil.github.io/deeplearning/filter-visualizations) [[ref2]](https://blog.keras.io/how-convolutional-neural-networks-see-the-world.html)
The helpful [video](https://www.youtube.com/watch?v=FrKWiRv254g) you can check out too.

Thanks for your work.
Happy cooperation 😄
